aggregator/src/aggregator/queue.rs

```diff
@@ -0,0 +1,712 @@
+                depth: u8::arbitrary(g) as usize,
+                concurrency: u8::arbitrary(g) as u32 + 1,
+                requests: (u16::arbitrary(g) / 10) as usize + 1,
+            }
+        }
+    }
+
+    #[derive(Debug, Clone, Copy)]
+    enum RuntimeFlavor {
+        CurrentThread,
+        MultiThread,
+    }
+
+    impl RuntimeFlavor {
+        fn run<F: Future>(&self, future: F) -> F::Output {
+            match self {
+                RuntimeFlavor::CurrentThread => RuntimeBuilder::new_current_thread(),
+                RuntimeFlavor::MultiThread => RuntimeBuilder::new_multi_thread(),
+            }
+            .enable_all()
+            .build()
+            .unwrap()
+            .block_on(async move { timeout(TEST_TIMEOUT, future).await.unwrap() })
+        }
+    }
+
+    #[test]
+    fn quickcheck_lifo_concurrency() {
+        struct ConcurrencyAssertingHandler {
+            max_concurrency: u32,
+            concurrency: AtomicU32,
+        }
+
+        #[async_trait]
+        impl Handler for ConcurrencyAssertingHandler {
+            async fn run(&self, conn: trillium::Conn) -> trillium::Conn {
+                let concurrency = self.concurrency.fetch_add(1, Ordering::Relaxed);
+                assert!(concurrency < self.max_concurrency);
+
+                // Somewhat arbitrary yield point, to give the dispatcher a chance to signal
+                // another concurring future. This is mostly pertinent if we're running the test on
+                // a current_thread runtime. Otherwise, without await points, on a current_thread
+                // runtime, this function won't yield.
+                yield_now().await;
+
+                let conn = conn.ok("hello");
+                self.concurrency.fetch_sub(1, Ordering::Relaxed);
+                conn
+            }
+        }
+
+        fn qc(parameters: Parameters) {
+            debug!(?parameters, "quickcheck_lifo_concurrency parameters");
+            let Parameters {
+                runtime_flavor,
+                depth,
+                concurrency,
+                requests,
+            } = parameters;
+
+            runtime_flavor.run(async move {
+                let queue =
+                    Arc::new(LIFORequestQueue::new(concurrency, depth, &noop_meter()).unwrap());
+                let handler = Arc::new(queued_lifo(
+                    Arc::clone(&queue),
+                    ConcurrencyAssertingHandler {
+                        concurrency: Default::default(),
+                        max_concurrency: concurrency,
+                    },
+                ));
+
+                join_all((0..requests).map(|_| {
+                    let handler = Arc::clone(&handler);
+                    async move {
+                        get("/").run_async(&handler).await;
+                    }
+                }))
+                .await;
+            });
+        }
+
+        install_test_trace_subscriber();
+        quickcheck(qc as fn(Parameters));
```

Comment:
This style of invoking quickcheck is nicer, since it lets us definitively instantiate the trace subscriber once. This is especially helper for fiddling with the tokio-console, which is helpful for debugging hangs and is not permissive of multiple invocations of `console_subscriber::init()`.